stats/view/worker.go

```diff
@@ -129,6 +129,16 @@ func SetReportingPeriod(d time.Duration) {
 	<-req.c // don't return until the timer is set to the new duration.
 }
 
+// ReportUsage immediately reports collected data to all registered exporters
+// outside of the regular reporting cycle.
+func ReportUsage() {
```

Comment:
This is new API surface area so probably needs to be discussed on opencensus-specs first so that other language implementations can also be aware of it. I would recommend that for now we split this PR into two: one that just adds the flush-on-unregister behavior and another with the new API.